clients/admin-ui/src/features/privacy-notices/table/usePrivacyNoticesTable.tsx

```diff
@@ -0,0 +1,226 @@
+import { expandCollapseAllMenuItems } from "~/features/common/table/cells/constants";
+import { LinkCell } from "~/features/common/table/cells/LinkCell";
+import { useAntTable, useTableState } from "~/features/common/table/hooks";
+import { FRAMEWORK_MAP } from "~/features/privacy-notices/constants";
+import { useGetAllPrivacyNoticesQuery } from "~/features/privacy-notices/privacy-notices.slice";
+import MechanismCell from "~/features/privacy-notices/table/cells/MechanismCell";
+import NoticeChildrenCell from "~/features/privacy-notices/table/cells/NoticeChildrenCell";
+import NoticeEnableCell from "~/features/privacy-notices/table/cells/NoticeEnableCell";
+import StatusCell from "~/features/privacy-notices/table/cells/StatusCell";
+import { PrivacyNoticeRowType } from "~/features/privacy-notices/table/PrivacyNoticeRowType";
+import {
+  ConsentMechanism,
+  LimitedPrivacyNoticeResponseSchema,
+  PrivacyNoticeRegion,
+  ScopeRegistryEnum,
+} from "~/types/api";
+
+const EmptyTableNotice = () => {
+  const router = useRouter();
+
+  return (
+    <Empty
+      image={Empty.PRESENTED_IMAGE_SIMPLE}
+      description={
+        <Flex vertical gap="small">
+          <div>No privacy notices found.</div>
+          <div>
+            <Button
+              onClick={() => router.push(`${PRIVACY_NOTICES_ROUTE}/new`)}
+              type="primary"
+              size="small"
+              data-testid="add-privacy-notice-btn"
+            >
+              Add a privacy notice +
+            </Button>
+          </div>
+        </Flex>
+      }
+    />
+  );
+};
+
+const usePrivacyNoticesTable = () => {
+  const userCanUpdate = useHasPermission([
+    ScopeRegistryEnum.PRIVACY_NOTICE_UPDATE,
+  ]);
+
+  const tableState = useTableState({
+    pagination: {
+      defaultPageSize: 25,
+      pageSizeOptions: [25, 50, 100],
+    },
+  });
+
+  const { pageIndex, pageSize } = tableState;
+
+  // Fetch data
+  const { data, isLoading, isFetching } = useGetAllPrivacyNoticesQuery({
+    page: pageIndex,
+    size: pageSize,
+  });
+
+  const [isLocationsExpanded, setIsLocationsExpanded] = useState(false);
+
+  const items = useMemo(() => data?.items ?? [], [data?.items]);
+  const dataSource: PrivacyNoticeRowType[] = useMemo(
+    () =>
+      items.map((item) => {
+        const { children, ...rest } = item;
+        return {
+          ...rest,
+          noticeChildren: children,
+        };
+      }),
+    [items],
+  );
+
+  const totalRows = data?.total ?? 0;
+
+  const antTableConfig = useMemo(
+    () => ({
+      dataSource,
+      totalRows,
+      isLoading,
+      isFetching,
+      getRowKey: (record: PrivacyNoticeRowType) => record.id,
+      customTableProps: {
+        locale: {
+          emptyText: <EmptyTableNotice />,
+        },
+      },
+    }),
+    [totalRows, isLoading, isFetching, dataSource],
+  );
+
+  const { tableProps } = useAntTable(tableState, antTableConfig);
+
+  const columns: ColumnsType<PrivacyNoticeRowType> = useMemo(
```

Comment:
Instead of `PrivacyNoticeRowType`, you can just use `typeof dataSource` which presumably is the actual data being passed to the component.

```suggestion
  const columns: ColumnsType<typeof dataSource> = useMemo(
```